vars/mailSendNotification.groovy

```diff
@@ -0,0 +1,186 @@
+                    return
+                }
+                recipientList += getCulpritCommitters(config, script.currentBuild)
+            }
+            if(config.notificationRecipients)
+                recipientList +=  " ${config.notificationRecipients}"
+            emailext(
+                mimeType: 'text/html',
+                subject: subject.trim(),
+                body: mailContent,
+                to: recipientList.trim(),
+                recipientProviders: [requestor()],
+                attachLog: config.notificationAttachment
+            )
+        }
+    }
+}
+
+def getNumberOfCommits(buildList){
+    def numCommits = 0
+    if(buildList != null)
+        for(actBuild in buildList) {
+            def changeLogSets = actBuild.getChangeSets()
+            if(changeLogSets != null)
+                for(changeLogSet in changeLogSets)
+                    for(change in changeLogSet)
+                        numCommits++
+        }
+    return numCommits
+}
+
+def getCulpritCommitters(config, currentBuild) {
+    def recipients
+    def buildList = []
+    def build = currentBuild
+
+    if (build != null) {
+        // At least add the current build
+        buildList.add(build)
+
+        // Now collect FAILED or ABORTED ones
+        build = build.getPreviousBuild()
+        while (build != null) {
+            if (build.getResult() != 'SUCCESS') {
+                buildList.add(build)
+            } else {
+                break
+            }
+            build = build.getPreviousBuild()
+        }
+    }
+    def numberOfCommits = getNumberOfCommits(buildList)
+    if(config.wrapInNode){
+        node(){
+            try{
+                recipients = getCulprits(config, env.BRANCH_NAME, numberOfCommits)
+            }finally{
+                deleteDir()
+            }
+        }
+    }else{
+        recipients = getCulprits(config, env.BRANCH_NAME, numberOfCommits)
+        deleteDir()
+    }
+    echo "[${STEP_NAME}] last ${numberOfCommits} commits revealed following responsibles ${recipients}"
+    return recipients
+}
+
+def getCulprits(config, branch, numberOfCommits) {
+    if (branch?.startsWith('PR-')) {
+        deleteDir()
+        sshagent(
+            credentials: [config.gitSshKeyCredentialsId],
+            ignoreMissing: true
+        ) {
+            sh 'git init'
+            def pullRequestID = branch.replaceAll('PR-', '')
+            def localBranchName = "pr" + pullRequestID;
+            sh "git fetch ${config.gitUrl} pull/${pullRequestID}/head:${localBranchName} > /dev/null 2>&1"
+            sh "git checkout -f ${localBranchName} > /dev/null 2>&1"
+        }
+    } else {
+        if (config.gitCommitId) {
+            deleteDir()
+            sshagent(
+                credentials: [config.gitSshKeyCredentialsId],
+                ignoreMissing: true
+            ) {
+                sh "git clone ${config.gitUrl} ."
```

Comment:
The two git statements can be easily condensed into one.

Reply:
yepp